File: windows/Tama.Core/Mood.cs
namespace Tama.Core;

public enum MoodKind
{
    Greeting,   // transient: first activity of a new local day
    Working,    // a session is active right now (Intensity = active-session count)
    Resting,    // recent activity, but paused
    Napping,    // idle / away / nothing today
}

/// <summary>The companion cat's current feeling. Positive-only (no sad/sick states).</summary>
public readonly record struct Mood(MoodKind Kind, int Intensity = 0)
{
    public static readonly Mood Greeting = new(MoodKind.Greeting);
    public static readonly Mood Resting = new(MoodKind.Resting);
    public static readonly Mood Napping = new(MoodKind.Napping);
    public static Mood Working(int intensity) => new(MoodKind.Working, intensity);
}

/// <summary>The cat's own persisted state (NOT agent data): the last day it "saw" activity,
/// which gates the once-per-day greeting.</summary>
public sealed record CatState(DateTimeOffset? LastSeenDay)
{
    public static readonly CatState Initial = new((DateTimeOffset?)null);
}

File: Sources/TamaCore/Pet/Mood.swift
import Foundation

/// The companion cat's current feeling, derived from live activity. Positive-only:
/// there is no sad/sick/dying state and no needs-decay clock (see the Phase 1 spec).
public enum Mood: Sendable, Equatable {
    /// Transient: the first activity of a new local day.
    case greeting
    /// A session is active right now. `intensity` = count of currently-active sessions.
    case working(intensity: Int)
    /// Recent activity, but paused.
    case resting
    /// Idle / away / no activity today.
    case napping
}

/// The cat's own persisted state (NOT agent data). Phase 1 stores only the last day the
/// cat "saw" activity, which gates the once-per-day greeting.
public struct CatState: Sendable, Equatable, Codable {
    public var lastSeenDay: Date?
    public init(lastSeenDay: Date? = nil) { self.lastSeenDay = lastSeenDay }
    public static let initial = CatState()
}
